In [2]:
import numpy as np
import tensorflow as tf
from tensorflow import keras
import pandas as pd
from matplotlib import pyplot as plt
%matplotlib inline

In [9]:
df = pd.read_csv("MLA_File.csv")
df

b  Force_coef
0  0.10    0.003885
1  0.11    0.004385
2  0.12    0.004917
3  0.13    0.005503
4  0.14    0.006149
5  0.15    0.006973
6  0.16    0.007553
7  0.17    0.008382
8  0.18    0.009412
9  0.19    0.014717

In [10]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(df[['b']],df.Force_coef,test_size=0.2, random_state=25)

In [27]:
X_train_scaled = X_train.copy()
X_train_scaled['b'] = X_train_scaled['b'] / 100

X_test_scaled = X_test.copy()
X_test_scaled['b'] = X_test_scaled['b'] / 100

In [31]:
model = keras.Sequential([
    keras.layers.Dense(1, input_shape=(1,), activation='sigmoid', kernel_initializer='ones', bias_initializer='zeros')
])

model.compile(optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy'])

model.fit(X_train_scaled, y_train, epochs=5000)

Epoch 1/5000
1/1 [==============================] - 0s 283ms/step - loss: 0.6939 - accuracy: 0.0000e+00
Epoch 2/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.6934 - accuracy: 0.0000e+00
Epoch 3/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.6929 - accuracy: 0.0000e+00
Epoch 4/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.6924 - accuracy: 0.0000e+00
Epoch 5/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.6919 - accuracy: 0.0000e+00
Epoch 6/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.6914 - accuracy: 0.0000e+00
Epoch 7/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.6909 - accuracy: 0.0000e+00
Epoch 8/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.6904 - accuracy: 0.0000e+00
Epoch 9/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.6899 - accuracy: 0.0000e+00
Epoch 10/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.6894 

1/1 [==============================] - 0s 3ms/step - loss: 0.6202 - accuracy: 0.0000e+00
Epoch 160/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.6197 - accuracy: 0.0000e+00
Epoch 161/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.6193 - accuracy: 0.0000e+00
Epoch 162/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.6188 - accuracy: 0.0000e+00
Epoch 163/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.6184 - accuracy: 0.0000e+00
Epoch 164/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.6180 - accuracy: 0.0000e+00
Epoch 165/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.6175 - accuracy: 0.0000e+00
Epoch 166/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.6171 - accuracy: 0.0000e+00
Epoch 167/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.6167 - accuracy: 0.0000e+00
Epoch 168/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.616

1/1 [==============================] - 0s 3ms/step - loss: 0.5558 - accuracy: 0.0000e+00
Epoch 316/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.5554 - accuracy: 0.0000e+00
Epoch 317/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.5550 - accuracy: 0.0000e+00
Epoch 318/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.5547 - accuracy: 0.0000e+00
Epoch 319/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.5543 - accuracy: 0.0000e+00
Epoch 320/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.5539 - accuracy: 0.0000e+00
Epoch 321/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.5535 - accuracy: 0.0000e+00
Epoch 322/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.5531 - accuracy: 0.0000e+00
Epoch 323/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.5527 - accuracy: 0.0000e+00
Epoch 324/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.552

1/1 [==============================] - 0s 3ms/step - loss: 0.4991 - accuracy: 0.0000e+00
Epoch 472/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.4987 - accuracy: 0.0000e+00
Epoch 473/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.4984 - accuracy: 0.0000e+00
Epoch 474/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.4980 - accuracy: 0.0000e+00
Epoch 475/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4977 - accuracy: 0.0000e+00
Epoch 476/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.4974 - accuracy: 0.0000e+00
Epoch 477/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4970 - accuracy: 0.0000e+00
Epoch 478/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.4967 - accuracy: 0.0000e+00
Epoch 479/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.4963 - accuracy: 0.0000e+00
Epoch 480/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.496

1/1 [==============================] - 0s 3ms/step - loss: 0.4491 - accuracy: 0.0000e+00
Epoch 628/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4488 - accuracy: 0.0000e+00
Epoch 629/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.4485 - accuracy: 0.0000e+00
Epoch 630/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.4482 - accuracy: 0.0000e+00
Epoch 631/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4479 - accuracy: 0.0000e+00
Epoch 632/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4476 - accuracy: 0.0000e+00
Epoch 633/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4473 - accuracy: 0.0000e+00
Epoch 634/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4470 - accuracy: 0.0000e+00
Epoch 635/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.4467 - accuracy: 0.0000e+00
Epoch 636/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.446

1/1 [==============================] - 0s 2ms/step - loss: 0.4051 - accuracy: 0.0000e+00
Epoch 784/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.4048 - accuracy: 0.0000e+00
Epoch 785/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.4046 - accuracy: 0.0000e+00
Epoch 786/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.4043 - accuracy: 0.0000e+00
Epoch 787/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.4040 - accuracy: 0.0000e+00
Epoch 788/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4038 - accuracy: 0.0000e+00
Epoch 789/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4035 - accuracy: 0.0000e+00
Epoch 790/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.4032 - accuracy: 0.0000e+00
Epoch 791/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.4030 - accuracy: 0.0000e+00
Epoch 792/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.402

1/1 [==============================] - 0s 4ms/step - loss: 0.3664 - accuracy: 0.0000e+00
Epoch 940/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.3661 - accuracy: 0.0000e+00
Epoch 941/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3659 - accuracy: 0.0000e+00
Epoch 942/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.3657 - accuracy: 0.0000e+00
Epoch 943/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3654 - accuracy: 0.0000e+00
Epoch 944/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.3652 - accuracy: 0.0000e+00
Epoch 945/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.3650 - accuracy: 0.0000e+00
Epoch 946/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3647 - accuracy: 0.0000e+00
Epoch 947/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.3645 - accuracy: 0.0000e+00
Epoch 948/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.364

1/1 [==============================] - 0s 3ms/step - loss: 0.3323 - accuracy: 0.0000e+00
Epoch 1096/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3321 - accuracy: 0.0000e+00
Epoch 1097/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.3318 - accuracy: 0.0000e+00
Epoch 1098/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.3316 - accuracy: 0.0000e+00
Epoch 1099/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3314 - accuracy: 0.0000e+00
Epoch 1100/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3312 - accuracy: 0.0000e+00
Epoch 1101/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.3310 - accuracy: 0.0000e+00
Epoch 1102/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.3308 - accuracy: 0.0000e+00
Epoch 1103/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.3306 - accuracy: 0.0000e+00
Epoch 1104/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 4ms/step - loss: 0.3024 - accuracy: 0.0000e+00
Epoch 1251/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3022 - accuracy: 0.0000e+00
Epoch 1252/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3020 - accuracy: 0.0000e+00
Epoch 1253/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3018 - accuracy: 0.0000e+00
Epoch 1254/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.3016 - accuracy: 0.0000e+00
Epoch 1255/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3015 - accuracy: 0.0000e+00
Epoch 1256/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.3013 - accuracy: 0.0000e+00
Epoch 1257/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.3011 - accuracy: 0.0000e+00
Epoch 1258/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.3009 - accuracy: 0.0000e+00
Epoch 1259/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 2ms/step - loss: 0.2758 - accuracy: 0.0000e+00
Epoch 1407/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2756 - accuracy: 0.0000e+00
Epoch 1408/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2754 - accuracy: 0.0000e+00
Epoch 1409/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2753 - accuracy: 0.0000e+00
Epoch 1410/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.2751 - accuracy: 0.0000e+00
Epoch 1411/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2750 - accuracy: 0.0000e+00
Epoch 1412/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.2748 - accuracy: 0.0000e+00
Epoch 1413/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.2747 - accuracy: 0.0000e+00
Epoch 1414/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2745 - accuracy: 0.0000e+00
Epoch 1415/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 4ms/step - loss: 0.2522 - accuracy: 0.0000e+00
Epoch 1563/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2521 - accuracy: 0.0000e+00
Epoch 1564/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2520 - accuracy: 0.0000e+00
Epoch 1565/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2518 - accuracy: 0.0000e+00
Epoch 1566/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.2517 - accuracy: 0.0000e+00
Epoch 1567/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2515 - accuracy: 0.0000e+00
Epoch 1568/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2514 - accuracy: 0.0000e+00
Epoch 1569/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2512 - accuracy: 0.0000e+00
Epoch 1570/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2511 - accuracy: 0.0000e+00
Epoch 1571/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.2314 - accuracy: 0.0000e+00
Epoch 1719/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2312 - accuracy: 0.0000e+00
Epoch 1720/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.2311 - accuracy: 0.0000e+00
Epoch 1721/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2310 - accuracy: 0.0000e+00
Epoch 1722/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2309 - accuracy: 0.0000e+00
Epoch 1723/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2307 - accuracy: 0.0000e+00
Epoch 1724/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2306 - accuracy: 0.0000e+00
Epoch 1725/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2305 - accuracy: 0.0000e+00
Epoch 1726/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.2304 - accuracy: 0.0000e+00
Epoch 1727/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 2ms/step - loss: 0.2128 - accuracy: 0.0000e+00
Epoch 1875/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2127 - accuracy: 0.0000e+00
Epoch 1876/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2126 - accuracy: 0.0000e+00
Epoch 1877/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2125 - accuracy: 0.0000e+00
Epoch 1878/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2124 - accuracy: 0.0000e+00
Epoch 1879/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2123 - accuracy: 0.0000e+00
Epoch 1880/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2122 - accuracy: 0.0000e+00
Epoch 1881/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.2120 - accuracy: 0.0000e+00
Epoch 1882/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.2119 - accuracy: 0.0000e+00
Epoch 1883/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 4ms/step - loss: 0.1963 - accuracy: 0.0000e+00
Epoch 2031/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1962 - accuracy: 0.0000e+00
Epoch 2032/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1961 - accuracy: 0.0000e+00
Epoch 2033/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1960 - accuracy: 0.0000e+00
Epoch 2034/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1959 - accuracy: 0.0000e+00
Epoch 2035/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1958 - accuracy: 0.0000e+00
Epoch 2036/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.1957 - accuracy: 0.0000e+00
Epoch 2037/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1956 - accuracy: 0.0000e+00
Epoch 2038/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1955 - accuracy: 0.0000e+00
Epoch 2039/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 2ms/step - loss: 0.1816 - accuracy: 0.0000e+00
Epoch 2187/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1815 - accuracy: 0.0000e+00
Epoch 2188/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1814 - accuracy: 0.0000e+00
Epoch 2189/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1813 - accuracy: 0.0000e+00
Epoch 2190/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1812 - accuracy: 0.0000e+00
Epoch 2191/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1811 - accuracy: 0.0000e+00
Epoch 2192/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1810 - accuracy: 0.0000e+00
Epoch 2193/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1809 - accuracy: 0.0000e+00
Epoch 2194/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1808 - accuracy: 0.0000e+00
Epoch 2195/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.1684 - accuracy: 0.0000e+00
Epoch 2342/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1684 - accuracy: 0.0000e+00
Epoch 2343/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1683 - accuracy: 0.0000e+00
Epoch 2344/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1682 - accuracy: 0.0000e+00
Epoch 2345/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1681 - accuracy: 0.0000e+00
Epoch 2346/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1680 - accuracy: 0.0000e+00
Epoch 2347/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1680 - accuracy: 0.0000e+00
Epoch 2348/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1679 - accuracy: 0.0000e+00
Epoch 2349/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1678 - accuracy: 0.0000e+00
Epoch 2350/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 5ms/step - loss: 0.1567 - accuracy: 0.0000e+00
Epoch 2497/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1566 - accuracy: 0.0000e+00
Epoch 2498/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1565 - accuracy: 0.0000e+00
Epoch 2499/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1565 - accuracy: 0.0000e+00
Epoch 2500/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1564 - accuracy: 0.0000e+00
Epoch 2501/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1563 - accuracy: 0.0000e+00
Epoch 2502/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1563 - accuracy: 0.0000e+00
Epoch 2503/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1562 - accuracy: 0.0000e+00
Epoch 2504/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1561 - accuracy: 0.0000e+00
Epoch 2505/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.1461 - accuracy: 0.0000e+00
Epoch 2653/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1460 - accuracy: 0.0000e+00
Epoch 2654/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1459 - accuracy: 0.0000e+00
Epoch 2655/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1459 - accuracy: 0.0000e+00
Epoch 2656/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1458 - accuracy: 0.0000e+00
Epoch 2657/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1457 - accuracy: 0.0000e+00
Epoch 2658/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1457 - accuracy: 0.0000e+00
Epoch 2659/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1456 - accuracy: 0.0000e+00
Epoch 2660/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1455 - accuracy: 0.0000e+00
Epoch 2661/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.1365 - accuracy: 0.0000e+00
Epoch 2809/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1364 - accuracy: 0.0000e+00
Epoch 2810/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1364 - accuracy: 0.0000e+00
Epoch 2811/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1363 - accuracy: 0.0000e+00
Epoch 2812/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1362 - accuracy: 0.0000e+00
Epoch 2813/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1362 - accuracy: 0.0000e+00
Epoch 2814/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1361 - accuracy: 0.0000e+00
Epoch 2815/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1361 - accuracy: 0.0000e+00
Epoch 2816/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1360 - accuracy: 0.0000e+00
Epoch 2817/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.1279 - accuracy: 0.0000e+00
Epoch 2965/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1278 - accuracy: 0.0000e+00
Epoch 2966/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1277 - accuracy: 0.0000e+00
Epoch 2967/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1277 - accuracy: 0.0000e+00
Epoch 2968/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1276 - accuracy: 0.0000e+00
Epoch 2969/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1276 - accuracy: 0.0000e+00
Epoch 2970/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1275 - accuracy: 0.0000e+00
Epoch 2971/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.1275 - accuracy: 0.0000e+00
Epoch 2972/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1274 - accuracy: 0.0000e+00
Epoch 2973/5000
1/1 [==============================] - 0s 2ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.1201 - accuracy: 0.0000e+00
Epoch 3121/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1200 - accuracy: 0.0000e+00
Epoch 3122/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1200 - accuracy: 0.0000e+00
Epoch 3123/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1199 - accuracy: 0.0000e+00
Epoch 3124/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1199 - accuracy: 0.0000e+00
Epoch 3125/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1198 - accuracy: 0.0000e+00
Epoch 3126/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1198 - accuracy: 0.0000e+00
Epoch 3127/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1197 - accuracy: 0.0000e+00
Epoch 3128/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1197 - accuracy: 0.0000e+00
Epoch 3129/5000
1/1 [==============================] - 0s 2ms/step - lo

1/1 [==============================] - 0s 5ms/step - loss: 0.1130 - accuracy: 0.0000e+00
Epoch 3277/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1130 - accuracy: 0.0000e+00
Epoch 3278/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1129 - accuracy: 0.0000e+00
Epoch 3279/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1129 - accuracy: 0.0000e+00
Epoch 3280/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1128 - accuracy: 0.0000e+00
Epoch 3281/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1128 - accuracy: 0.0000e+00
Epoch 3282/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1128 - accuracy: 0.0000e+00
Epoch 3283/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1127 - accuracy: 0.0000e+00
Epoch 3284/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1127 - accuracy: 0.0000e+00
Epoch 3285/5000
1/1 [==============================] - 0s 2ms/step - lo

1/1 [==============================] - 0s 2ms/step - loss: 0.1066 - accuracy: 0.0000e+00
Epoch 3433/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1066 - accuracy: 0.0000e+00
Epoch 3434/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1065 - accuracy: 0.0000e+00
Epoch 3435/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1065 - accuracy: 0.0000e+00
Epoch 3436/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.1065 - accuracy: 0.0000e+00
Epoch 3437/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1064 - accuracy: 0.0000e+00
Epoch 3438/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1064 - accuracy: 0.0000e+00
Epoch 3439/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1064 - accuracy: 0.0000e+00
Epoch 3440/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1063 - accuracy: 0.0000e+00
Epoch 3441/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.1009 - accuracy: 0.0000e+00
Epoch 3588/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1008 - accuracy: 0.0000e+00
Epoch 3589/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1008 - accuracy: 0.0000e+00
Epoch 3590/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1008 - accuracy: 0.0000e+00
Epoch 3591/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1007 - accuracy: 0.0000e+00
Epoch 3592/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1007 - accuracy: 0.0000e+00
Epoch 3593/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.1007 - accuracy: 0.0000e+00
Epoch 3594/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.1006 - accuracy: 0.0000e+00
Epoch 3595/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.1006 - accuracy: 0.0000e+00
Epoch 3596/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 4ms/step - loss: 0.0956 - accuracy: 0.0000e+00
Epoch 3743/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0956 - accuracy: 0.0000e+00
Epoch 3744/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0956 - accuracy: 0.0000e+00
Epoch 3745/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0955 - accuracy: 0.0000e+00
Epoch 3746/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0955 - accuracy: 0.0000e+00
Epoch 3747/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0955 - accuracy: 0.0000e+00
Epoch 3748/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0955 - accuracy: 0.0000e+00
Epoch 3749/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0954 - accuracy: 0.0000e+00
Epoch 3750/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0954 - accuracy: 0.0000e+00
Epoch 3751/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.0909 - accuracy: 0.0000e+00
Epoch 3898/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0909 - accuracy: 0.0000e+00
Epoch 3899/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0908 - accuracy: 0.0000e+00
Epoch 3900/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0908 - accuracy: 0.0000e+00
Epoch 3901/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0908 - accuracy: 0.0000e+00
Epoch 3902/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0907 - accuracy: 0.0000e+00
Epoch 3903/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0907 - accuracy: 0.0000e+00
Epoch 3904/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0907 - accuracy: 0.0000e+00
Epoch 3905/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0907 - accuracy: 0.0000e+00
Epoch 3906/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.0865 - accuracy: 0.0000e+00
Epoch 4054/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0865 - accuracy: 0.0000e+00
Epoch 4055/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0865 - accuracy: 0.0000e+00
Epoch 4056/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0865 - accuracy: 0.0000e+00
Epoch 4057/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0864 - accuracy: 0.0000e+00
Epoch 4058/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0864 - accuracy: 0.0000e+00
Epoch 4059/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0864 - accuracy: 0.0000e+00
Epoch 4060/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0863 - accuracy: 0.0000e+00
Epoch 4061/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0863 - accuracy: 0.0000e+00
Epoch 4062/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.0826 - accuracy: 0.0000e+00
Epoch 4210/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0825 - accuracy: 0.0000e+00
Epoch 4211/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0825 - accuracy: 0.0000e+00
Epoch 4212/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0825 - accuracy: 0.0000e+00
Epoch 4213/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0825 - accuracy: 0.0000e+00
Epoch 4214/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0824 - accuracy: 0.0000e+00
Epoch 4215/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0824 - accuracy: 0.0000e+00
Epoch 4216/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0824 - accuracy: 0.0000e+00
Epoch 4217/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.0824 - accuracy: 0.0000e+00
Epoch 4218/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 2ms/step - loss: 0.0790 - accuracy: 0.0000e+00
Epoch 4366/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0789 - accuracy: 0.0000e+00
Epoch 4367/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0789 - accuracy: 0.0000e+00
Epoch 4368/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0789 - accuracy: 0.0000e+00
Epoch 4369/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0789 - accuracy: 0.0000e+00
Epoch 4370/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0788 - accuracy: 0.0000e+00
Epoch 4371/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0788 - accuracy: 0.0000e+00
Epoch 4372/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0788 - accuracy: 0.0000e+00
Epoch 4373/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0788 - accuracy: 0.0000e+00
Epoch 4374/5000
1/1 [==============================] - 0s 2ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.0757 - accuracy: 0.0000e+00
Epoch 4521/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0757 - accuracy: 0.0000e+00
Epoch 4522/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0756 - accuracy: 0.0000e+00
Epoch 4523/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0756 - accuracy: 0.0000e+00
Epoch 4524/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0756 - accuracy: 0.0000e+00
Epoch 4525/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0756 - accuracy: 0.0000e+00
Epoch 4526/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0756 - accuracy: 0.0000e+00
Epoch 4527/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0755 - accuracy: 0.0000e+00
Epoch 4528/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0755 - accuracy: 0.0000e+00
Epoch 4529/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.0727 - accuracy: 0.0000e+00
Epoch 4677/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0727 - accuracy: 0.0000e+00
Epoch 4678/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0726 - accuracy: 0.0000e+00
Epoch 4679/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0726 - accuracy: 0.0000e+00
Epoch 4680/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0726 - accuracy: 0.0000e+00
Epoch 4681/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0726 - accuracy: 0.0000e+00
Epoch 4682/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0726 - accuracy: 0.0000e+00
Epoch 4683/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0726 - accuracy: 0.0000e+00
Epoch 4684/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0725 - accuracy: 0.0000e+00
Epoch 4685/5000
1/1 [==============================] - 0s 4ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.0700 - accuracy: 0.0000e+00
Epoch 4832/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0699 - accuracy: 0.0000e+00
Epoch 4833/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0699 - accuracy: 0.0000e+00
Epoch 4834/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0699 - accuracy: 0.0000e+00
Epoch 4835/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0699 - accuracy: 0.0000e+00
Epoch 4836/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0699 - accuracy: 0.0000e+00
Epoch 4837/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0699 - accuracy: 0.0000e+00
Epoch 4838/5000
1/1 [==============================] - 0s 4ms/step - loss: 0.0698 - accuracy: 0.0000e+00
Epoch 4839/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0698 - accuracy: 0.0000e+00
Epoch 4840/5000
1/1 [==============================] - 0s 3ms/step - lo

1/1 [==============================] - 0s 3ms/step - loss: 0.0675 - accuracy: 0.0000e+00
Epoch 4987/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0675 - accuracy: 0.0000e+00
Epoch 4988/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0674 - accuracy: 0.0000e+00
Epoch 4989/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0674 - accuracy: 0.0000e+00
Epoch 4990/5000
1/1 [==============================] - 0s 5ms/step - loss: 0.0674 - accuracy: 0.0000e+00
Epoch 4991/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0674 - accuracy: 0.0000e+00
Epoch 4992/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0674 - accuracy: 0.0000e+00
Epoch 4993/5000
1/1 [==============================] - 0s 2ms/step - loss: 0.0674 - accuracy: 0.0000e+00
Epoch 4994/5000
1/1 [==============================] - 0s 3ms/step - loss: 0.0674 - accuracy: 0.0000e+00
Epoch 4995/5000
1/1 [==============================] - 0s 2ms/step - lo

In [32]:
model.evaluate(X_test_scaled,y_test)

1/1 [==============================] - 0s 64ms/step - loss: 0.0614 - accuracy: 0.0000e+00


[0.06140384078025818, 0.0]

In [33]:
X_test

b
5  0.15
1  0.11

In [34]:
y_test

5    0.006973
1    0.004385
Name: Force_coef, dtype: float64

In [35]:
model.predict(X_test_scaled)

array([[0.04279688],
       [0.04283121]], dtype=float32)

here the value after doing prediction is not matching with the actual value . 
means it requires a lot of changes in the NURAL NETWORK 

In [25]:
coef, intercept = model.get_weights()

In [26]:
coef, intercept

(array([[-1.9247072]], dtype=float32), array([-2.943835], dtype=float32))

HERE THE value for w1 = -1.9247 and value for b = -2.943835

now doing the calculation without tensor flow 

In [36]:
def sigmoid(x):
        import math
        return 1 / (1 + math.exp(-x))
sigmoid(18)

0.9999999847700205

In [42]:
def prediction_function(b):
    weighted_sum = coef[0]*b +intercept
    return sigmoid(weighted_sum)

prediction_function(0.15)

0.03795934163337895

In [43]:
def sigmoid_numpy(X):
   return 1/(1+np.exp(-X))

In [44]:
def log_loss(y_true, y_predicted):
    epsilon = 1e-15
    y_predicted_new = [max(i,epsilon) for i in y_predicted]
    y_predicted_new = [min(i,1-epsilon) for i in y_predicted_new]
    y_predicted_new = np.array(y_predicted_new)
    return -np.mean(y_true*np.log(y_predicted_new)+(1-y_true)*np.log(1-y_predicted_new))

In [47]:
def gradient_descent(b, y_true, epochs, loss_thresold):
    w1 = 1
    bias = 0
    rate = 0.5
    n = len(b)
    for i in range(epochs):
        weighted_sum = w1 * b + bias
        y_predicted = sigmoid_numpy(weighted_sum)
        loss = log_loss(y_true, y_predicted)

        w1d = (1/n)*np.dot(np.transpose(b),(y_predicted-y_true)) 

        bias_d = np.mean(y_predicted-y_true)
        w1 = w1 - rate * w1d
       
        bias = bias - rate * bias_d

        print (f'Epoch:{i}, w1:{w1}, bias:{bias}, loss:{loss}')

        if loss<=loss_thresold:
            break

    return w1, bias

In [53]:
gradient_descent(X_train_scaled['b'],y_train,5000, 0.0673)

Epoch:0, w1:0.9996338868507628, bias:-0.24640349802425915, loss:0.6938791195147639
Epoch:1, w1:0.9993133633375308, bias:-0.4621586113000873, loss:0.5800201871377783
Epoch:2, w1:0.9990317002013238, bias:-0.6517893667291412, loss:0.49259125927111613
Epoch:3, w1:0.9987826920675654, bias:-0.819467524325142, loss:0.4248699455325349
Epoch:4, w1:0.9985610016561098, bias:-0.9687812007278621, loss:0.37174744066634424
Epoch:5, w1:0.9983621987736769, bias:-1.1027086764146254, loss:0.3294795027781232
Epoch:6, w1:0.9981826663371948, bias:-1.2236815872222015, loss:0.29535966749364784
Epoch:7, w1:0.9980194675986056, bias:-1.333674193499271, loss:0.2674330443941853
Epoch:8, w1:0.9978702166237017, bias:-1.4342904580313327, loss:0.24427823373973667
Epoch:9, w1:0.9977329668949352, bias:-1.526838937538705, loss:0.22485082496669992
Epoch:10, w1:0.9976061207362764, bias:-1.6123936748484566, loss:0.20837388403373122
Epoch:11, w1:0.9974883575986982, bias:-1.6918424109137045, loss:0.19426190266125049
Epoch:12,

(0.995363446361168, -3.1322084182578593)

In [1]:
coef, intercept

NameError: name 'coef' is not defined